services/src/main/java/io/druid/cli/RouterJettyServerInitializer.java

```diff
@@ -72,18 +77,19 @@ public void initialize(Server server, Injector injector)
     final ServletContextHandler root = new ServletContextHandler(ServletContextHandler.SESSIONS);
 
     root.addServlet(new ServletHolder(new DefaultServlet()), "/*");
-    root.addServlet(
-        new ServletHolder(
-            new AsyncQueryForwardingServlet(
-                jsonMapper,
-                smileMapper,
-                hostFinder,
-                httpClient,
-                emitter,
-                requestLogger
-            )
-        ), "/druid/v2/*"
+
+    final AsyncQueryForwardingServlet asyncQueryForwardingServlet = new AsyncQueryForwardingServlet(
+        jsonMapper,
+        smileMapper,
+        hostFinder,
+        httpClientProvider,
+        httpClientConfig,
+        emitter,
+        requestLogger
     );
+    asyncQueryForwardingServlet.setTimeout(httpClientConfig.getReadTimeout().getMillis());
+
+    root.addServlet(new ServletHolder(asyncQueryForwardingServlet), "/druid/v2/*");
     root.addFilter(defaultAsyncGzipFilterHolder(), "/*", null);
     // Can't use '/*' here because of Guice conflicts with AsyncQueryForwardingServlet path
     root.addFilter(GuiceFilter.class, "/status/*", null);
```

Comment:
Also, I think it'd be worthwhile to test a few queries through /druid/v2/. I've seen unexpected behavior with Servlet chains in jetty before